### PR TITLE
Rename shrink -> simplify

### LINKED ISSUE #36: Rename Shrinker -> Simplifier?
Hi,

May I give a suggestion: rename the `Shrinker` type to `Simplifier`.

This is a suggestion that occurred to me whilst reading through #33 but I thought it worth making a new issue rather than derailing that one.

Following the (de-facto?) principle of using descriptive names in elm, I thought that the meaning of "shrinking a test input" would have been much clearer to me if it had instead been called "simplifying a test input".

From @mgold's comment: https://github.com/elm-explorations/test/issues/33#issuecomment-425282488

> Let's say I have a test that claims every integer is even. You'll get around 50 failing tests, each with a different odd number. The goal of the shrinker is to tell you that 1 is the "smallest" or "simplest" counterexample to your claim. 1 is ~~"larger"~~ "smaller" than -1, somewhat by convention.

Feel free to close if such a change is not appropriate, or if github is not the appropriate place to make such a suggestion.

Cheers :+1:

### PR DESCRIPTION
**Pull request by harrysarson**
This PR caries out the renaming suggested in #36. It also adds aliases to ensure that no functions are removed the API.

Due to the way `elm diff` works, this PR is still a MAJOR change. The `Shrinker a` type is changed from a type alias of `a -> LazyList a` to a type alias of `Simplify.Simplifier a`. This change should cause no problems in practice but elm considers it to be a MAJOR change. (There is a similar change made in `Test.Runner`.)

This could be made into a MINOR change but keeping the `Shrinker` type as is and making `Simplify.Simplifier` a type alias to it but I believe that will make the docs confusing so it is worth making a major change.

I also updated the docs written in #35 - if @Bastes is able to review 9309a73 that would be ace.

---

closes #36

### CODE REVIEW

File: elm.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
         "Expect",
         "Fuzz",
         "Shrink",
```
[the remark this answers is not in the document]
Comment (harrysarson) [reply elm.json]: I readded it in  8bf2f6b, if we decide not to keep the aliases we can revert that commit and shrink will be removed from the list.

File: src/Fuzz.elm [hunk 2]
```diff
@@ -426,13 +426,13 @@ listShrinkRecurse listOfTrees =
             else
                 Lazy.List.empty
 
-        shrinkOne prefix aList =
+        simplifyOne prefix aList =
             case aList of
                 [] ->
                     Lazy.List.empty
 
                 (Rose x shrunkenXs) :: more ->
```
[the remark this answers is not in the document]
Comment (harrysarson) [reply src/Fuzz.elm]: Updated now :)

**Comment by harrysarson** [comment]
I have updated elm-format and repushed my changes, I also addressed some of @drathier and @Bastes comments.

If we do not want to keep `Shrink` etc around then I can remove.

**Comment by drathier** [comment]
Do we want to keep the backwards compatible `Shrink` module around? Next release is a major release anyway. Emojivote up for "keep the Shrink module" and down for "drop the Shrink module".

**Comment by harrysarson** [comment]
I guess the vote says remove `Shrink` then?

**Comment by Bastes** [comment]
Looks like it :) (watch gleefully as you sharpen your knives ;p)

**Comment by drathier** [comment]
Yep, I've cast my vote as well now. Let's drop it :)

**Comment by mgold** [comment]
Sounds good to me. Gang, before releasing a new major version let's make sure everyone has a chance to review all the changes.

**Comment by harrysarson** [comment]
I have removed the `Shrink` aliases. From my end this is good to go.

**Comment by harrysarson** [comment]
What is the status here?

**Comment by drathier** [comment]
Great work, this is sure to help beginners learn how <s>shrinking</s> simplifying works 🎉

**Comment by harrysarson** [comment]
🎉

**Comment by Bastes** [comment]
Late to the party, but :tada: :)

**Pull request merged**